chi/server.py

```diff
@@ -85,211 +89,300 @@ def instance_create_args(
 
-    @property
-    def error(self) -> bool:
-        """Check if the instance is in an error state."""
-        return self.status == "ERROR"
+        server = cls(name=nova_server.name,
+                     reservation_id=None,
+                     image_name = get_image(image_id).name,
+                     flavor_name=get_flavor(flavor_id).name,
+                     key_name=nova_server.key_name,
+                     network_name=get_network(network_id)['name'])
 
-    def wait(self):
-        """Wait for the server instance to finish launching.
+        try:
+            created_at = nova_server.created
+        except Exception:
+            created_at = nova_server.created_at
 
-        If the server goes into an error state, this function will return early.
-        """
-        self.conn.compute.wait_for_server(self.server, wait=(60 * 20))
+        try:
+            host_id = nova_server.hostId
+        except Exception:
+            host_id = nova_server.host_id
 
-    def associate_floating_ip(self):
-        """Attach a floating IP to this server instance."""
-        if self.ip is not None:
-            return self.ip
+        try:
+            host_status = nova_server.host_status
+        except Exception:
+            host_status = None
 
-        self._fip, self._fip_created = get_or_create_floating_ip()
-        self.ip = self._fip["floating_ip_address"]
-        self.conn.compute.add_floating_ip_to_server(self.server.id, self.ip)
-        return self.ip
+        try:
+            hypervisor_hostname = nova_server.hypervisor_hostname
+        except Exception:
+            hypervisor_hostname = None
 
-    def disassociate_floating_ip(self):
-        """Detach the floating IP attached to this server instance, if any."""
-        if self.ip is None:
-            return
+        try:
+            is_locked = nova_server.is_locked
+        except Exception:
+            is_locked = None
+
+        server.id = nova_server.id
+        server._status = nova_server.status
+        server._addresses = nova_server.addresses
+        server.created_at = created_at
+        server.host_id = host_id
+        server.host_status = host_status
+        server.hypervisor_hostname = hypervisor_hostname
+        server.is_locked = is_locked
 
-        self.conn.compute.remove_floating_ip_from_server(
-            self.server.id, self.ip)
-        if self._fip_created:
-            self.neutron.delete_floatingip(self._fip["id"])
+        return server
 
-        self.ip = None
-        self._fip = None
-        self._fip_created = False
+    def delete(self) -> None:
+        delete_server(self.id)
 
-    def delete(self):
-        """Delete this server instance."""
-        self.conn.compute.delete_server(self.server)
-        self.conn.compute.wait_for_delete(self.server)
+    def wait(self, status: str = "ACTIVE") -> None:
+        print(f"Waiting for server {self.name}'s status to turn to {status}. This can take up to 12 minutes")
```

Comment:
We should timeout at 30 minutes, I would say typically this can range from 8-18 minutes. CascadelakeR nodes are slower than skylake.